sha1_mb/sha1_mb_x4_avx.asm

```diff
@@ -389,14 +389,13 @@ lloop:
 	vmovdqa [ARG1 + 4*16], E
 
 	; update input pointers
-	add     inp0, IDX
-	mov     [ARG1 + _data_ptr + 0*8], inp0
-	add     inp1, IDX
-	mov     [ARG1 + _data_ptr + 1*8], inp1
-	add     inp2, IDX
-	mov     [ARG1 + _data_ptr + 2*8], inp2
-	add     inp3, IDX
-	mov     [ARG1 + _data_ptr + 3*8], inp3
+	vmovq	xmm1, IDX
+	vpbroadcastq xmm1, xmm1
```

Comment:
This is an AVX2 instruction, so unfortunately, this breaks architectures older than Ivy Bridge (same in other _avx implementations). You'll probably need to use vpinsrq, although I don't expect much improvement here. Having said this, we are planning on removing all the _avx implementations anyway, keeping _sse, _avx2 and _avx512, so I wouldn't bother making changes in the _avx path.